framework/src/actions/SplitMeshAction.C

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SplitMeshAction.h"
+
+#include "MooseApp.h"
+#include "MooseUtils.h"
+#include "MooseMesh.h"
+#include "libmesh/checkpoint_io.h"
+
+template <>
+InputParameters
+validParams<SplitMeshAction>()
+{
+  return validParams<Action>();
+}
+
+SplitMeshAction::SplitMeshAction(InputParameters params) : Action(params) {}
+
+void
+SplitMeshAction::act()
+{
+  auto mesh = _app.actionWarehouse().mesh();
+
+  if (mesh->getFileName() == "" && _app.parameters().get<std::string>("split_file") == "")
+    mooseError("must specify an output mesh file name (--split-file) when splitting "
```

Comment:
I know this isn't in the style guide but, we typically write sentences (start with a capital letter) in all of our error messaging throughout MOOSE - please change.